leases/management/commands/import_winter_leases_from_csv.py

```diff
@@ -0,0 +1,171 @@
+            type=str,
+            help="[Required] The API token for Profile",
+        )
+        parser.add_argument(
+            "--lease-file-path",
+            nargs="?",
+            type=str,
+            help="[Required] The file to parse",
+        )
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            default=False,
+            help="Only show a list of the invoices that can be sent "
+            "and how many customer are missing phones in our system",
+        )
+
+    def get_or_create_customer_profile(
+        self, name: str, email: str = None, phone: str = None
+    ) -> CustomerProfile:
+        names = name.split(" ")
+        first_name = names.pop().capitalize()
+        last_name = " ".join([name.capitalize() for name in names])
+
+        try:
+            helsinki_profile = self.profile_service.find_profile(
+                first_name, last_name, email, phone, force_only_one=True
+            )
+            profile = CustomerProfile.objects.get(id=helsinki_profile.id)
+        except (NoProfilesException, CustomerProfile.DoesNotExist):
+            logger.debug(f"Creating profile for {last_name}, {first_name} ({email})")
+            profile = self.profile_service.create_profile(
+                first_name, last_name, email, phone
+            )
+        return profile
+
+    def handle(
+        self, *args, profile_token=None, lease_file_path=None, dry_run=False, **options
+    ):
+        self.profile_service = ProfileService(profile_token=profile_token)
+
+        multiple_profiles = []
+        successful = []
+        failed = []
+        OTHER_BOAT_TYPE = BoatType.objects.get(id=9)
+
+        with open(lease_file_path, "r", encoding="utf-8-sig") as cf:
+            rd = csv.reader(cf, delimiter=";")
+            for line in rd:
+                lease_input = LeaseInput(*line)
+                try:
+                    customer_profile = self.get_or_create_customer_profile(
+                        lease_input.name,
+                        email=lease_input.email,
+                        phone=lease_input.phone,
+                    )
+                except MultipleProfilesException as e:
+                    multiple_profiles.append(
+                        (
+                            lease_input.section_id,
+                            lease_input.place_number,
+                            lease_input.name,
+                            ";".join(e.ids),
+                        )
+                    )
+                    continue
+
+                try:
+                    place = WinterStoragePlace.objects.get(
+                        winter_storage_section_id=lease_input.section_id,
+                        number=lease_input.place_number,
+                    )
+
+                    boat = None
+                    if lease_input.boat_width and lease_input.boat_length:
+                        boat, _created = Boat.objects.get_or_create(
```

Comment:
What if admins already have created a Boat for the user? Maybe skip get_or_create step if the customer already has a boat with the same registration number?

Reply:
Added first a check if the customer has a matching boat :thu